cloud-platform/app-deploy.md

```diff
@@ -0,0 +1,332 @@
+
+See the contents of the `ingress.yml` file below:
+
+```Yaml
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: cp-demo-app-ing
+spec:
+  rules:
+  - host: cp-demo-app.integration.dsd.io
+    http:
+      paths:
+      - path: /
+        backend:
+          serviceName: cp-demo-app-svc
+          servicePort: 80
+```
+
+If you are using the file as a template for your own application, replace the `cp-demo-app-ing` tag with one suited to your application.
+
+Also, ensure the `serviceName:` tag is the same as specified in the `service.yml`.
+
+## Configuring secrets
+
+The following is an example of encoding (configuring) aws access-key credentials in your deployment.
+(See https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables
+for detailed information regarding providing base64 values in secret objects to Kuberenetes pods)
+
+Create your AWS Credentials access key (making a note of the aws_access_key_id and aws_secret_access_key)
+
+put the aws key secret and id into a file (in this example demo-v9.k8s) as follows ('thekey' referencing the aws_access_key_id
+and 'thesecret' referencing the aws_secret_access_key):
+
+```
+apiVersion: v1
+
+kind: Secret
+
+metadata:
+
+  name: demosecret-v9
+
+type: Opaque
+
+data:
+
+  thekey: AKIAFTKSAW15HJLOGD
+
+  thesecret: g8hjpmhvgfhk4547gfdshhjj
+
+```
+
+create secret generic secret from that file:
+```$kubectl -n demo-app create secret generic demosecret-v9 --from-file=demosecret-v9=demo-v9.k8s```
+
+'demo-app' being the namespace, 'demosecret-v9' referencing the 'name' in demo-v9.k8s and referencing the file 'demo-v9.k8s'
+this will give the following screen output
+
+
+```
+secret "demosecret-v9" created
+```
+To view the output of your secret:
+
+```
+$kubectl -o json -n demo-app get secret demosecret-v9
+
+```
+
+This will output something like the following:
+
+```
+{
+    "apiVersion": "v1",
+    "data": {
+        "demosecret-v9": "YXBpVmVyc2lvbjogdjEKa2luZDogU2VjcmV0Cm1ldGFkYXRhOgogIG5hbWU6IGRlbW9zZWNyZXQtdjkKdHlwZTogT3BhcXVlCmRhdGE6CiAgdGhla2V5OiBBS0lBRlRLU0FXMTVISkxPR0QKICB0aGVzZWNyZXQ6IGc4aGpwbWh2Z2ZoazQ1NDdnZmRzaGhqago="
+    },
+    "kind": "Secret",
+    "metadata": {
+        "creationTimestamp": "2018-05-09T15:17:50Z",
+        "name": "demosecret-v9",
+        "namespace": "demo-app",
+        "resourceVersion": "3021923",
+        "selfLink": "/api/v1/namespaces/demo-app/secrets/demosecret-v9",
+        "uid": "2315be2f-539c-11e8-8ff4-0aecd97fec6e"
+    },
+    "type": "Opaque"
+}
+
+```
+Using the 'data' output from above, issue the following command (to validate and verify that it is the same as you originally inputted into the demo-v9.k8s file above):
+
+```
+$ base64 -D
+YXBpVmVyc2lvbjogdjEKa2luZDogU2VjcmV0Cm1ldGFkYXRhOgogIG5hbWU6IGRlbW9zZWNyZXQtdjkKdHlwZTogT3BhcXVlCmRhdGE6CiAgdGhla2V5OiBBS0lBRlRLU0FXMTVISkxPR0QKICB0aGVzZWNyZXQ6IGc4aGpwbWh2Z2ZoazQ1NDdnZmRzaGhqago=
+
+```
+ctrl+d to exit
```

Comment:
This will terminate the user's shell session - I don't think we want them to do that.